src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -601,11 +601,16 @@ MavlinkMissionManager::handle_mission_ack(const mavlink_message_t *msg)
 	if (CHECK_SYSID_COMPID_MISSION(wpa)) {
 		if ((msg->sysid == _transfer_partner_sysid && msg->compid == _transfer_partner_compid)) {
 			if (_state == MAVLINK_WPM_STATE_SENDLIST && _mission_type == wpa.mission_type) {
+
 				_time_last_recv = hrt_absolute_time();
 
-				if (_transfer_seq == current_item_count()) {
+				if (wpa.type == MAV_MISSION_ACCEPTED && _transfer_seq == current_item_count()) {
 					PX4_DEBUG("WPM: MISSION_ACK OK all items sent, switch to state IDLE");
 
+				} else if (wpa.type == MAV_MISSION_OPERATION_CANCELLED) {
+					PX4_DEBUG("WPM: MISSION_ACK CANCELLED, switch to state IDLE");
+					_transfer_in_progress = false;
```

Comment:
This is only set for mission uploads, whereas here we land only on downloads, so you don't need to reset it here.

Reply:
You're right but that's not obvious from the name.
Removed in dd505a9240.

Reply:
You remove the wrong one.

Reply:
Stupid me, thanks. ff23988ed9